src/Compilers/Core/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -73,6 +73,30 @@ public static int Main(string[] args)
 
             var pipeName = args[0].Substring(pipeArgPrefix.Length);
 
+            var serverMutexName = $"{pipeName}.server";
+            bool holdsMutex;
+            using (var serverMutex = new Mutex(initiallyOwned: true,
+                                               name: serverMutexName,
+                                               createdNew: out holdsMutex))
+            {
+                if (!holdsMutex)
+                {
+                    return CommonCompiler.Failed;
```

Comment:
I wonder if it would be useful to expand the CommonCompiler enumeration to list the different kinds of "failed".
In this case it's because another server is already running on the given pipe. Above, it's because the pipename was missing or mal-formed.